middle_end/flambda/flambda_to_clambda.ml

```diff
@@ -70,22 +73,29 @@ let is_function_constant t closure_id =
 (* Instrumentation of closure and field accesses to try to catch compiler
    bugs. *)
 
-let check_closure ulam named : Clambda.ulambda =
+let check_closure t ulam named : Clambda.ulambda =
   if not !Clflags.clambda_checks then ulam
   else
     let desc =
       Primitive.simple ~name:"caml_check_value_is_closure"
         ~arity:2 ~alloc:false
     in
     let str = Format.asprintf "%a" Flambda.print_named named in
-    let str_const =
+    let sym =
       Compilenv.new_structured_constant (Uconst_string str) ~shared:true
```

Comment:
This doesn't just create a symbol name, it also records the structured constant. This should probably be replaced by 'new_const_symbol'

Or at least, flambda should use `clear_structured_constants` if this is used.

Reply:
Fixed